crates/primitives/chain_config/src/rpc_version.rs

```diff
@@ -31,39 +26,42 @@ impl RpcVersion {
-        let parts: Vec<&str> = path.split('/').collect();
-
-        log::debug!(target: "rpc_version", "version parts are: {parts:?}");
+        let parts: Vec<&str> = path.split('/').filter(|part| !part.is_empty()).collect();
 
-        // If we have an empty path or just "/", fallback to latest rpc version
-        if parts.len() == 1 || (parts.len() == 2 && parts[1].is_empty()) {
-            log::debug!(target: "rpc_version", "no version, defaulting to latest");
-            return Ok(Self::RPC_VERSION_LATEST);
-        }
+        tracing::debug!(target: "rpc_version", "version parts are: {parts:?}");
 
-        // Check if the path follows the correct format, i.e. /rpc/v[version].
-        // If not, fallback to the latest version
-        if parts.len() != 3 || parts[1] != "rpc" || !parts[2].starts_with('v') {
-            log::debug!(target: "rpc_version", "invalid version format, defaulting to latest");
-            return Ok(Self::RPC_VERSION_LATEST);
-        }
-
-        log::debug!(target: "rpc_version", "looking for matching version...");
-        let version_str = &parts[2][1..]; // without the 'v' prefix
-        if let Ok(version) = RpcVersion::from_str(version_str) {
-            if SUPPORTED_RPC_VERSIONS.contains(&version) {
-                log::debug!(target: "rpc_version", "found matching version: {version}");
-                Ok(version)
-            } else {
-                log::debug!(target: "rpc_version", "no matching version");
-                Err(RpcVersionError::UnsupportedVersion)
+        match parts.as_slice() {
+            // Match empty path
+            [] => {
+                tracing::debug!(target: "rpc_version", "No version specified, defaulting to latest.");
+                Ok(Self::RPC_VERSION_LATEST)
+            }
+            // Match valid path format "/rpc/vX_Y_Z"
+            ["rpc", version_str] if version_str.starts_with('v') => {
+                let version_str = &version_str[1..]; // strip "v"
+                match RpcVersion::from_str(version_str) {
+                    Ok(version) if SUPPORTED_RPC_VERSIONS.contains(&version) => {
+                        tracing::debug!(target: "rpc_version", "Found supported version: {version}");
+                        Ok(version)
+                    }
+                    Ok(_) => {
+                        tracing::debug!(target: "rpc_version", "Version unsupported");
+                        Err(RpcVersionError::UnsupportedVersion)
+                    }
+                    Err(_) => {
+                        tracing::debug!(target: "rpc_version", "Invalid version format: {version_str}");
+                        Err(RpcVersionError::InvalidVersion)
+                    }
+                }
+            }
+            // Fallback for invalid format
+            _ => {
+                tracing::debug!(target: "rpc_version", "Invalid path format, defaulting to latest.");
+                Ok(Self::RPC_VERSION_LATEST)
```

Comment:
Should we not also return `RpcVersionError::InvalidVersion` in this case?

Reply:
I just refactoring